lib/data/context.js

```diff
@@ -0,0 +1,50 @@
+const isGenerator = require('../utils').isGenerator;
+const DataTable = require('./table');
+
+module.exports = function (context) {
+
+  context.Data = function (dataTable) {
+    let data = detectDataType(dataTable);
+
+    return {
+      Scenario: function (title, opts, fn) {
+        if (typeof opts === 'function' && !fn) {
+          fn = opts;
+          opts = {};
+        }
+        data.forEach((dataRow) => {
+          opts.data = dataRow;
+          let test = context.Scenario(`${title} | ${dataRow}`, opts, fn);
+          test.inject.current = dataRow;
```

Comment:
now `current` is not able to use as pageObject name. We have to change pageObject validation here https://github.com/Codeception/CodeceptJS/blob/master/lib/container.js#L142

to block using `current` and allow `scenarioContext`